app/controllers/users_controller.rb

```diff
@@ -1399,6 +1400,24 @@ def revoke_auth_token
     render json: success_json
   end
 
+  def feature_topic
+    user = fetch_user_from_params
+    guardian.ensure_can_edit!(user)
+    topic = Topic.find(params[:topic_id].to_i)
+
```

Comment:
We're going to need more checks on the topic here. I suggest a `can_feature_topic` guardian method because we'll want to make sure:

1. The topic is 100% public (not a PM, not in a secure group)
2. The user created the topic

Otherwise I could hit this API endpoint and feature a topic that is private!